proto/config/fw.proto

```diff
@@ -10,6 +10,46 @@ option java_package = "org.lfedge.eve.config";
+  //           for outbound and inbound flow if the destination and the source IP address
+  //           matches the given value, respectively. Can be combined with any other match
+  //           type to further narrow down the selection criteria.
+  //   * "host": value should be a domain name of a remote endpoint. It can be either a fully
+  //             qualified, or a partially qualified domain name (FQDN or PQDN). A packet is
+  //             matched if it is destined to or originated from an IP address that was obtained
+  //             by a DNS query for that exact domain or any of its subdomains. For example,
+  //             match of type "host" with value "domain.com" will also apply to the endpoint
+  //             "subdomain.domain.com". Can be combined with other match types except for "eidset".
+  //   * "eidset": special match type for the overlay network. Matches IPs of all applications
+  //               deployed in the same network as well as all IPs with statically configured
+  //               DNS entries (under the config field NetworkInstanceConfig.Dns). For this type,
+  //               value field is not used. Can be combined with other match types except for
+  //               "host".
+  //   * "protocol": value should specify the protocol to match. Protocol can be one of "tcp",
+  //                 "udp", "icmp", or "all", or it can be a numeric value, representing one
+  //                 of these protocols or a different one. A protocol name from /etc/protocols
+  //                 is also allowed. Protocol match can be combined with any other match type
+  //                 (often combined with port numbers).
+  //   * "lport": value should be an application local port number. For filtering actions,
+  //              this is the source port for outbound traffic and destination port for inbound
+  //              traffic. For PORTMAP action, this represents application port as exposed
+  //              to the external network (i.e., if <edge-node-ip>:2222 is mapped to <app-ip>:22,
+  //              lport refers to 2222). lport can be combined with any other match type.
+  //              It is actually required to combine "lport" and "protocol" inside the same ACE.
+  //              In other words, port without protocol is not valid.
+  //   * "fport": value should be a remote endpoint port number (foreign port). Used for filtering
+  //              actions, but not for PORTMAP (do not confuse with "lport", which is still used to
+  //              represent the forwarded port - the forwarded port is still considered as local).
+  //              "fport" can be combined with any other match type. It is actually required
+  //              to combine "fport" with "protocol" inside the same ACE. In other words, port
+  //              without protocol is not valid.
+  //  * "adapter": value should be an adapter label (SystemAdapter.Labels). It can be used for
```

Comment:
Your example used "eth0" i.e., an interface name (or logical label?).
Assuming we want that capability we can document that a port always has an implicit label which is its (logical label, ifname, pick one?).

Reply:
Since application config (and fw rules) can be in theory reused for multiple edge devices (like the commercial controller does), the `adapter` value must be label and cannot reference interface name.
In my example I have label `uplink`: https://github.com/milan-zededa/eve-api/blob/adapter-labels/proto/config/fw.proto#L81
I think you are referencing the example for the network instance port, where the interface name is allowed: https://github.com/milan-zededa/eve-api/blob/adapter-labels/proto/config/netinst.proto#L125

Reply:
L125 is no longer correct but it's been months. Were you referring to https://github.com/milan-zededa/eve-api/blob/0ee94e298044c9e7299db0b30d64d161887d1016/proto/config/netinst.proto#L176?